addons/wowl/static/src/components/dropdown/dropdown.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, core, hooks, useState } from "@odoo/owl";
+import { OwlEvent } from "@odoo/owl/dist/types/core/owl_event";
+import { DropdownItemClickedPayload, ParentClosingMode } from "./dropdown_item";
+
+export class Dropdown extends Component {
+  static bus = new core.EventBus();
+  static props = {
+    open: {
+      type: Boolean,
+      optional: true,
+    },
+    menuClass: {
+      type: String,
+      optional: true,
+    },
+    togglerClass: {
+      type: String,
+      optional: true,
+    },
+  };
+  static template = "wowl.Dropdown";
+
+  state = useState({ open: this.props.open, groupIsOpen: false });
+
+  constructor() {
+    super(...arguments);
+    // Close on outside click listener
+    hooks.useExternalListener(window, "click", this.onWindowClicked);
+    // Listen to siblings state
+    Dropdown.bus.on("state-changed", this, this.onSiblingDropdownStateChanged);
+  }
+
+  // --------------------------------------------------------------------------
+  // PRIVATE
+  // --------------------------------------------------------------------------
+  private _changeStateAndNotify(stateSlice: any) {
+    // Update the state
+    Object.assign(this.state, stateSlice);
+    // Notify over the bus
+    Dropdown.bus.trigger("state-changed", {
+      emitter: this,
+      newState: { ...this.state }
+    });
+  }
+  private _close() {
+    this._changeStateAndNotify({ open: false, groupIsOpen: false });
+  }
+  private _open() {
+    this._changeStateAndNotify({ open: true, groupIsOpen: true });
+  }
+  private _toggle() {
+    const toggled = !this.state.open;
+    this._changeStateAndNotify({
+      open: toggled,
+      groupIsOpen: toggled,
+    });
+  }
+
+  // --------------------------------------------------------------------------
+  // HANDLERS
+  // --------------------------------------------------------------------------
+  onItemClicked(ev: OwlEvent<DropdownItemClickedPayload>) {
+    // Handle parent closing request
+    const { dropdownClosingRequest } = ev.detail;
+    const closeAll = dropdownClosingRequest.mode === ParentClosingMode.AllParents;
+    const closeMe =
+      dropdownClosingRequest.isFresh &&
+      dropdownClosingRequest.mode === ParentClosingMode.ClosestParent;
+    if (closeAll || closeMe) {
+      this._close();
+    }
+    // Mark closing request as started
+    ev.detail.dropdownClosingRequest.isFresh = false;
```

Comment:
Comment should be a bit more clear, took my a good 30 sec to get it. Maybe the isFresh is not perfect, hasBeenConsumed feels more like the right term.

Reply:
Disagree, I can not see a better term than "the request is fresh". We then mark the request as not fresh once it has been treated at least once.

'hasBeenConsumed' vehiculate the idea that it is done and therefore implies that no more treatment will happen.